### ISSUE TITLE
Maven fails to build risev2g.evcc under Ubuntu 16.04.2 LTS (amd64)

### ISSUE THREAD

**Issue by lategoodbye** [V2GClarity/RISE-V2G#3]
If i try to build risev2g under Ubuntu 16.04 (amd64) i will always get a BUILD FAILURE:

`Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:2.4.1:single (make-my-jar-with-dependencies) on project evcc: Error reading assemblies: Error locating assembly descriptor: file:/home/stefan/RISE-V2G/RISE-V2G-EVCC/src/assembly/bin.xml`

Here some information about the environment:

```
stefan@Latitude-E4310:~/RISE-V2G$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="16.04.2 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.2 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```
```
stefan@Latitude-E4310:~/risev2g$ java -version
openjdk version "1.8.0_131"
OpenJDK Runtime Environment (build 1.8.0_131-8u131-b11-0ubuntu1.16.04.2-b11)
OpenJDK 64-Bit Server VM (build 25.131-b11, mixed mode)
```
```
stefan@Latitude-E4310:~/risev2g$ mvn -v
Apache Maven 3.3.9
Maven home: /usr/share/maven
Java version: 1.8.0_131, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-8-openjdk-amd64/jre
Default locale: de_DE, platform encoding: UTF-8
OS name: "linux", version: "4.10.9-041009-generic", arch: "amd64", family: "unix"
```

Finally the complete build output:

```
stefan@Latitude-E4310:~/RISE-V2G$ mvn package
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for org.eclipse.risev2g:evcc:jar:1.0.0-SNAPSHOT
[WARNING] 'build.plugins.plugin.(groupId:artifactId)' must be unique but found duplicate declaration of plugin org.apache.maven.plugins:maven-assembly-plugin @ org.eclipse.risev2g:evcc:[unknown-version], /home/stefan/RISE-V2G/RISE-V2G-EVCC/pom.xml, line 60, column 12
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for org.eclipse.risev2g:secc:jar:1.0.0-SNAPSHOT
[WARNING] 'build.plugins.plugin.(groupId:artifactId)' must be unique but found duplicate declaration of plugin org.apache.maven.plugins:maven-assembly-plugin @ org.eclipse.risev2g:secc:[unknown-version], /home/stefan/RISE-V2G/RISE-V2G-SECC/pom.xml, line 49, column 12
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] risev2g.parent
[INFO] risev2g.shared
[INFO] risev2g.evcc
[INFO] risev2g.secc
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building risev2g.parent 1.0.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building risev2g.shared 1.0.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[WARNING] The POM for net.sourceforge.openexi:nagasena:jar:0000.0002.0052.0 is missing, no dependency information available
[WARNING] The POM for net.sourceforge.openexi:nagasena-rta:jar:0000.0002.0052.0 is missing, no dependency information available
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ shared ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] Copying 8 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.2:compile (default-compile) @ shared ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ shared ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] skip non existing resourceDirectory /home/stefan/RISE-V2G/RISE-V2G-Shared/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.2:testCompile (default-testCompile) @ shared ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-surefire-plugin:2.17:test (default-test) @ shared ---
[INFO] No tests to run.
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ shared ---
[INFO] 
[INFO] --- maven-shade-plugin:2.4.1:shade (default) @ shared ---
[INFO] Including com.siemens.ct.exi:exificient:jar:0.9.6 in the shaded jar.
[INFO] Including com.siemens.ct.exi:exificient-core:jar:0.9.6 in the shaded jar.
[INFO] Including com.siemens.ct.exi:exificient-grammars:jar:0.9.6 in the shaded jar.
[INFO] Including xerces:xercesImpl:jar:2.11.0 in the shaded jar.
[INFO] Including xml-apis:xml-apis:jar:1.4.01 in the shaded jar.
[INFO] Including org.apache.logging.log4j:log4j-api:jar:2.1 in the shaded jar.
[INFO] Including org.apache.logging.log4j:log4j-core:jar:2.1 in the shaded jar.
[INFO] Including net.sourceforge.openexi:nagasena:jar:0000.0002.0052.0 in the shaded jar.
[INFO] Including net.sourceforge.openexi:nagasena-rta:jar:0000.0002.0052.0 in the shaded jar.
[WARNING] shared-1.0.0-SNAPSHOT.jar, exificient-0.9.6.jar define 31 overlapping classes: 
[WARNING]   - com.siemens.ct.exi.api.sax.SAXDecoder$DocTypeTextLexicalHandler
[WARNING]   - com.siemens.ct.exi.util.SkipRootElementXMLReader
[WARNING]   - com.siemens.ct.exi.api.dom.SaxToDomHandler
[WARNING]   - com.siemens.ct.exi.api.stream.StAXDecoder
[WARNING]   - com.siemens.ct.exi.cmd.CmdOption
[WARNING]   - com.siemens.ct.exi.util.SimpleDocTypeParser
[WARNING]   - com.siemens.ct.exi.api.stream.StAXEncoder
[WARNING]   - com.siemens.ct.exi.api.stream.EmptyLocation
[WARNING]   - com.siemens.ct.exi.api.stream.StAXDecoder$1
[WARNING]   - com.siemens.ct.exi.api.stream.StAXEncoder$EncoderNamespaceContext
[WARNING]   - 21 more...
[WARNING] shared-1.0.0-SNAPSHOT.jar, log4j-core-2.1.jar define 554 overlapping classes: 
[WARNING]   - org.apache.logging.log4j.core.config.xml.XmlConfiguration$Status
[WARNING]   - org.apache.logging.log4j.core.appender.ConsoleAppender$ConsoleManagerFactory
[WARNING]   - org.apache.logging.log4j.core.appender.AbstractManager
[WARNING]   - org.apache.logging.log4j.core.net.server.UdpSocketServer
[WARNING]   - org.apache.logging.log4j.core.async.AsyncLoggerContextSelector
[WARNING]   - org.apache.logging.log4j.core.config.plugins.visitors.PluginElementVisitor
[WARNING]   - org.apache.logging.log4j.core.appender.rolling.RollingFileManager$1
[WARNING]   - org.apache.logging.log4j.core.net.SslSocketManager$1
[WARNING]   - org.apache.logging.log4j.core.appender.db.jpa.JpaAppender
[WARNING]   - org.apache.logging.log4j.core.pattern.LineSeparatorPatternConverter
[WARNING]   - 544 more...
[WARNING] shared-1.0.0-SNAPSHOT.jar, exificient-core-0.9.6.jar define 208 overlapping classes: 
[WARNING]   - com.siemens.ct.exi.types.AbstractTypeCoder
[WARNING]   - com.siemens.ct.exi.grammars.event.StartElement
[WARNING]   - com.siemens.ct.exi.io.channel.BitEncoderChannel
[WARNING]   - com.siemens.ct.exi.grammars.event.DocType
[WARNING]   - com.siemens.ct.exi.core.EXIBodyDecoderReordered$EndElementEntry
[WARNING]   - com.siemens.ct.exi.attributes.AttributeFactory
[WARNING]   - com.siemens.ct.exi.grammars.event.StartDocument
[WARNING]   - com.siemens.ct.exi.core.AbstractEXIBodyEncoder$ProfileDisablingMechanism
[WARNING]   - com.siemens.ct.exi.values.ValueType
[WARNING]   - com.siemens.ct.exi.grammars.event.Comment
[WARNING]   - 198 more...
[WARNING] shared-1.0.0-SNAPSHOT.jar, xercesImpl-2.11.0.jar define 952 overlapping classes: 
[WARNING]   - org.apache.xerces.parsers.AbstractSAXParser
[WARNING]   - org.apache.xerces.xni.grammars.XMLGrammarDescription
[WARNING]   - org.apache.xml.serialize.SerializerFactoryImpl
[WARNING]   - org.apache.wml.WMLTdElement
[WARNING]   - org.apache.xerces.impl.io.Latin1Reader
[WARNING]   - org.apache.xerces.dom.DOMXSImplementationSourceImpl
[WARNING]   - org.apache.xerces.impl.dv.xs.Base64BinaryDV
[WARNING]   - org.apache.xerces.parsers.XIncludeAwareParserConfiguration
[WARNING]   - org.apache.xml.serialize.HTMLdtd
[WARNING]   - org.apache.xerces.dom.ElementDefinitionImpl
[WARNING]   - 942 more...
[WARNING] shared-1.0.0-SNAPSHOT.jar, xml-apis-1.4.01.jar define 346 overlapping classes: 
[WARNING]   - org.w3c.dom.css.CSSValueList
[WARNING]   - org.w3c.dom.ProcessingInstruction
[WARNING]   - org.w3c.dom.html.HTMLBodyElement
[WARNING]   - javax.xml.validation.SecuritySupport$6
[WARNING]   - javax.xml.transform.SecuritySupport$3
[WARNING]   - org.w3c.dom.Entity
[WARNING]   - org.w3c.dom.ls.LSParserFilter
[WARNING]   - org.w3c.dom.html.HTMLMetaElement
[WARNING]   - javax.xml.stream.events.ProcessingInstruction
[WARNING]   - javax.xml.transform.stax.StAXResult
[WARNING]   - 336 more...
[WARNING] shared-1.0.0-SNAPSHOT.jar, log4j-api-2.1.jar define 81 overlapping classes: 
[WARNING]   - org.apache.logging.log4j.message.LocalizedMessage
[WARNING]   - org.apache.logging.log4j.message.LocalizedMessageFactory
[WARNING]   - org.apache.logging.log4j.message.MessageFormatMessageFactory
[WARNING]   - org.apache.logging.log4j.ThreadContext$ContextStack
[WARNING]   - org.apache.logging.log4j.util.Strings
[WARNING]   - org.apache.logging.log4j.message.StructuredDataId
[WARNING]   - org.apache.logging.log4j.status.StatusData
[WARNING]   - org.apache.logging.log4j.message.LoggerNameAwareMessage
[WARNING]   - org.apache.logging.log4j.spi.LoggerContextFactory
[WARNING]   - org.apache.logging.log4j.simple.SimpleLoggerContext
[WARNING]   - 71 more...
[WARNING] shared-1.0.0-SNAPSHOT.jar, nagasena-rta-0000.0002.0052.0.jar define 66 overlapping classes: 
[WARNING]   - org.openexi.scomp.ProtoGrammar
[WARNING]   - org.openexi.scomp.EXISchemaFactory$GrammarLocation
[WARNING]   - org.openexi.scomp.EventTypeCache$4
[WARNING]   - org.openexi.scomp.EventWildcardNS
[WARNING]   - org.openexi.scomp.EXISchemaFactoryExceptionXMsg
[WARNING]   - org.openexi.scomp.EventTypeCache$1
[WARNING]   - org.openexi.scomp.EXISchemaFactory$ComparableAttributeUse
[WARNING]   - org.openexi.scomp.EXISchemaReader$SchemaScannerFacade
[WARNING]   - org.openexi.scomp.EventTypeCache$8
[WARNING]   - com.thaiopensource.exi.xsd.regex.jdk1_4.Translator$Subtraction
[WARNING]   - 56 more...
[WARNING] shared-1.0.0-SNAPSHOT.jar, nagasena-0000.0002.0052.0.jar define 203 overlapping classes: 
[WARNING]   - org.openexi.proc.EXIDecoder
[WARNING]   - org.openexi.proc.io.BitPackedScanner
[WARNING]   - org.openexi.proc.io.IntegerValueScriber$IntegerValue
[WARNING]   - org.openexi.proc.grammars.FragmentGrammar
[WARNING]   - org.openexi.proc.io.compression.ScannerChannelFactory
[WARNING]   - org.openexi.proc.grammars.BuiltinGrammar
[WARNING]   - org.openexi.proc.io.SimpleScanner$EXIEventSchemaMixedCharacters
[WARNING]   - org.openexi.proc.common.EventTypeList$1
[WARNING]   - org.openexi.proc.io.GYearValueScriber
[WARNING]   - org.openexi.proc.common.StringTable$1
[WARNING]   - 193 more...
[WARNING] shared-1.0.0-SNAPSHOT.jar, exificient-grammars-0.9.6.jar define 19 overlapping classes: 
[WARNING]   - com.siemens.ct.exi.grammars.persistency.GrammarsPreperation
[WARNING]   - com.siemens.ct.exi.grammars.EXIContentModelBuilder$XSAttributeUseSort
[WARNING]   - com.siemens.ct.exi.grammars.persistency.GrammarIdDispenser$1
[WARNING]   - com.siemens.ct.exi.grammars.persistency.GrammarIdDispenser
[WARNING]   - org.apache.xerces.impl.xpath.regex.EXIRegularExpression
[WARNING]   - com.siemens.ct.exi.grammars.persistency.Grammars2JSON$1
[WARNING]   - com.siemens.ct.exi.grammars.XSDGrammarsBuilder
[WARNING]   - com.siemens.ct.exi.grammars.EXIContentModelBuilder$XSAttributeDeclarationSort
[WARNING]   - com.siemens.ct.exi.grammars.persistency.Grammars2JSON
[WARNING]   - com.siemens.ct.exi.grammars.persistency.GrammarsPreperation$1
[WARNING]   - 9 more...
[WARNING] maven-shade-plugin has detected that some class files are
[WARNING] present in two or more JARs. When this happens, only one
[WARNING] single version of the class is copied to the uber jar.
[WARNING] Usually this is not harmful and you can skip these warnings,
[WARNING] otherwise try to manually exclude artifacts based on
[WARNING] mvn dependency:tree -Ddetail=true and the above output.
[WARNING] See http://docs.codehaus.org/display/MAVENUSER/Shade+Plugin
[INFO] Replacing original artifact with shaded artifact.
[INFO] Replacing /home/stefan/RISE-V2G/RISE-V2G-Shared/target/shared-1.0.0-SNAPSHOT.jar with /home/stefan/RISE-V2G/RISE-V2G-Shared/target/shared-1.0.0-SNAPSHOT-shaded.jar
[INFO] Dependency-reduced POM written at: /home/stefan/RISE-V2G/RISE-V2G-Shared/dependency-reduced-pom.xml
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building risev2g.evcc 1.0.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ evcc ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.2:compile (default-compile) @ evcc ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ evcc ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] skip non existing resourceDirectory /home/stefan/RISE-V2G/RISE-V2G-EVCC/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.2:testCompile (default-testCompile) @ evcc ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-surefire-plugin:2.17:test (default-test) @ evcc ---
[INFO] No tests to run.
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ evcc ---
[INFO] 
[INFO] --- maven-assembly-plugin:2.4.1:single (make-my-jar-with-dependencies) @ evcc ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] risev2g.parent ..................................... SUCCESS [  0.185 s]
[INFO] risev2g.shared ..................................... SUCCESS [  2.798 s]
[INFO] risev2g.evcc ....................................... FAILURE [  0.489 s]
[INFO] risev2g.secc ....................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 3.733 s
[INFO] Finished at: 2017-07-22T11:41:26+02:00
[INFO] Final Memory: 16M/246M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:2.4.1:single (make-my-jar-with-dependencies) on project evcc: Error reading assemblies: Error locating assembly descriptor: file:/home/stefan/RISE-V2G/RISE-V2G-EVCC/src/assembly/bin.xml
[ERROR] 
[ERROR] [1] [INFO] Searching for file location: /home/stefan/RISE-V2G/RISE-V2G-EVCC/file:/home/stefan/RISE-V2G/RISE-V2G-EVCC/src/assembly/bin.xml
[ERROR] 
[ERROR] [2] [INFO] File: /home/stefan/RISE-V2G/RISE-V2G-EVCC/file:/home/stefan/RISE-V2G/RISE-V2G-EVCC/src/assembly/bin.xml does not exist.
[ERROR] 
[ERROR] [3] [INFO] File: /home/stefan/RISE-V2G/file:/home/stefan/RISE-V2G/RISE-V2G-EVCC/src/assembly/bin.xml does not exist.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :evcc
```

**Comment by lategoodbye** [comment]
Not sure this is related but on Windows 10, maven 3.3.9, Oracle Java 8 SE i will get this warning during build:

```
[WARNING] Configuration options: 'appendAssemblyId' is set to false, and 'classifier' is missing.
Instead of attaching the assembly file: C:\Users\stefan\RISE-V2G-EVCC\target\evcc-1.0.0-SNAPSHOT.jar, it will become the file for main project artifact.
NOTE: If multiple descriptors or descriptor-formats are provided for this project, the value of this file will be non-deterministic!
[WARNING] Replacing pre-existing project main-artifact file: C:\Users\stefan\RISE-V2G-EVCC\target\evcc-1.0.0-SNAPSHOT.jar
with assembly file: C:\Users\stefan\RISE-V2G-EVCC\target\evcc-1.0.0-SNAPSHOT.jar
```

**Comment by MarcMueltin** [comment]
I am not a Maven expert to be honest. Will look into it when I have more time. Do you have any updates on this issue?

**Comment by lategoodbye** [comment]
This issue still persists. Currently i don't have the time for investigations. It seems to me a very trivial issue, because the protocol part "file:" is within the search string.

**Comment by mkupke** [comment]
In my Linux Ubuntu LTS environment the maven build is working fine. The difference to lategoodby environment is, that I'm using a bit newer version (some updates). Maybe this is your problem!?

```
mkupke@DE100454LX:~/V2GClarity/RISE-V2G$ java -version
java version "1.8.0_144"
Java(TM) SE Runtime Environment (build 1.8.0_144-b01)
Java HotSpot(TM) 64-Bit Server VM (build 25.144-b01, mixed mode)
```

```
mkupke@DE100454LX:~/V2GClarity/RISE-V2G$ mvn -version
Apache Maven 3.5.0 (ff8f5e7444045639af65f6095c62210b5713f426; 2017-04-03T21:39:06+02:00)
Maven home: /opt/apache-maven-3.5.0
Java version: 1.8.0_144, vendor: Oracle Corporation
Java home: /opt/jdk1.8.0_144/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.4.0-62-generic", arch: "amd64", family: "unix"
```

```
mkupke@DE100454LX:~/V2GClarity/RISE-V2G$ cat /etc/os-release
NAME="Ubuntu"
VERSION="16.04.3 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.3 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```

Hopefully this helps.
Best regards,
Martin Kupke

**Comment by lategoodbye** [comment]
Hi Martin,
i already updated to 16.04.3 but its still using Maven 3.3.9. It's possible that you are using non-default PPAs or installed Maven manually?

**Comment by mkupke** [comment]
I installed Maven manually to /opt and adapted PATH variable that mvn is found under /opt/apache-maven-3.5.0/bin
'''
wget http://www-eu.apache.org/dist/maven/maven-3/3.5.0/binaries/apache-maven-3.5.0-bin.tar.gz
tar -xvf apache-maven-3.5.0-bin.tar.gz
'''
Nothing "non-default", only standard things. I didn't install maven by package manager (e.g. apt), because following the link of Eclipse with the description and clicking on Maven I got to the website
http://maven.apache.org/download.cgi
and there is version 3.5.0 named as the current version. The package manager installed an older version.

**Comment by mkupke** [comment]
I installed Maven manually to /opt and adapted PATH variable that mvn is found under /opt/apache-maven-3.5.0/bin
'''
wget http://www-eu.apache.org/dist/maven/maven-3/3.5.0/binaries/apache-maven-3.5.0-bin.tar.gz
tar -xvf apache-maven-3.5.0-bin.tar.gz
'''
Nothing "non-default", only standard things. I didn't install maven by package manager (e.g. apt), because following the link of Eclipse with the description and clicking on Maven I got to the website
http://maven.apache.org/download.cgi
and there is version 3.5.0 named as the current version. The package manager installed an older version.

**Comment by lategoodbye** [comment]
Thanks. I also installed maven manually and issue doesn't occured anymore. This solution is acceptable for a developer, but not for "real life".

**Comment by MarcMueltin** [comment]
I committed a new version of RISE V2G: babd2ee
This update focuses a lot on Maven. The previous Maven files (pom.xml files) were not configured in a correct way. I spent a lot of time the past few days with reading through the Maven documentation and testing the final configuration on my Mac as well as on a Windows machine. Don't have Ubuntu installed yet.
I use Maven 3.5.0, but I think it should work with your older Maven version as well.

Please check out the latest version (it's not a release yet) and navigate to the RISE-V2G-Parent folder on your command line. Now in this folder, type "mvn clean install". This should produce the necessary .jar files for the SECC and the EVCC in the respective "/target" folders of each project, along with the keystores and .properties file.

Please let me know if this solves your problem.

**Comment by lategoodbye** [comment]
Wonderful, this fixed the issue (tested with Maven 3.3.9, Ubuntu 16.04.3).
